pkg/container/vector/vector.go

```diff
@@ -21,46 +21,39 @@ import (
 	"unsafe"
 
 	"github.com/matrixorigin/matrixone/pkg/common/bitmap"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/shuffle"
 
 	"github.com/matrixorigin/matrixone/pkg/common/moerr"
 	"github.com/matrixorigin/matrixone/pkg/common/mpool"
 	"github.com/matrixorigin/matrixone/pkg/container/nulls"
 	"github.com/matrixorigin/matrixone/pkg/container/types"
-	"github.com/matrixorigin/matrixone/pkg/vectorize/shuffle"
 )
 
-// XXX Moved vector from types.go to vector.go
-// XXX Deleted vector interface, which was commented out and outdated anyway.
-/* Vector vector
- * origin true:
- * 				count || type || bitmap size || bitmap || vector
- * origin false:
- *  			count || vector
- */
+const (
+	FLAT     = iota // flat vector represent a uncompressed vector
+	CONSTANT        // const vector
+	DIST            // dictionary vector
```

Comment:
Do we really need dist -- how much it really saves over varlena?   esp, long string varlena can points to one copy.

Reply:
I'm not sure. This may never need to be realized, leaving an opening in case it is needed in the future.